File: src/main/java/net/pearadise/pearexpansion/registry/FlammableBlocks.java
package net.pearadise.pearexpansion.registry;

import net.fabricmc.fabric.api.registry.FlammableBlockRegistry;
import net.pearadise.pearexpansion.block.ModBlocks;

public class FlammableBlocks {
    private static final int FLAME_ODDS = 5;
    private static final int BURN_ODDS = 20;

    public static void registerFlammableBlocks() {
        FlammableBlockRegistry registry = FlammableBlockRegistry.getDefaultInstance();

        registry.add(ModBlocks.OAK_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.SPRUCE_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.BIRCH_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.JUNGLE_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.ACACIA_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.DARK_OAK_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.MANGROVE_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.CHERRY_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.PALE_OAK_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.BAMBOO_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
        registry.add(ModBlocks.BAMBOO_MOSAIC_VERTICAL_SLAB, FLAME_ODDS, BURN_ODDS);
    }
}

File: src/main/kotlin/net/pearadise/pearexpansion/registry/FlammableBlocks.kt
package net.pearadise.pearexpansion.registry

import net.fabricmc.fabric.api.registry.FlammableBlockRegistry
import net.pearadise.pearexpansion.block.ModBlocks

/**
 * Registers flammable blocks for the mod.
 *
 * This object adds custom blocks to the flammable block registry,
 * allowing them to catch fire and burn in the game.
 */
object FlammableBlocks {

    /** Odds that fire will spread to the block. */
    private const val FLAME_ODDS = 5

    /** Odds that the block will burn once on fire. */
    private const val BURN_ODDS = 20

    /**
     * Registers all custom flammable blocks with the registry.
     *
     * Adds each block with defined flame and burn odds.
     */
    fun registerFlammableBlocks() {
        val registry = FlammableBlockRegistry.getDefaultInstance()

        // Add each vertical slab block to the flammable registry
        listOf(
            ModBlocks.OAK_VERTICAL_SLAB,
            ModBlocks.SPRUCE_VERTICAL_SLAB,
            ModBlocks.BIRCH_VERTICAL_SLAB,
            ModBlocks.JUNGLE_VERTICAL_SLAB,
            ModBlocks.ACACIA_VERTICAL_SLAB,
            ModBlocks.DARK_OAK_VERTICAL_SLAB,
            ModBlocks.MANGROVE_VERTICAL_SLAB,
            ModBlocks.CHERRY_VERTICAL_SLAB,
            ModBlocks.PALE_OAK_VERTICAL_SLAB,
            ModBlocks.BAMBOO_VERTICAL_SLAB,
            ModBlocks.BAMBOO_MOSAIC_VERTICAL_SLAB
        ).forEach { block ->
            registry.add(block, FLAME_ODDS, BURN_ODDS)
        }
    }
}